src/DGtal/images/ImageContainerBySTLMap.h

```diff
@@ -78,102 +79,109 @@ namespace DGtal
+   * stored in the underlying STL map) can be chosen by the user.
    *
-   * As a model of concepts::CImage, this class provides two ways of accessing values: 
-   * - through the range of points returned by the domain() method 
-   * combined with the operator() that takes a point and returns its associated value. 
-   * - through the range of values returned by the range() method, 
+   * As a model of concepts::CImage, this class provides two ways of accessing values:
+   * - through the range of points returned by the domain() method
+   * combined with the operator() that takes a point and returns its associated value.
+   * - through the range of values returned by the range() method,
    * which can be used to directly iterate over the values of the image
    *
-   * This class also provides a setValue() method and an output iterator, 
-   * which is returned by the outputIterator() method for writting purposes. 
+   * This class also provides a setValue() method and an output iterator,
+   * which is returned by the outputIterator() method for writting purposes.
    *
    * @see testImage.cpp
    */
 
   template <typename TDomain, typename TValue>
-  class ImageContainerBySTLMap: 
+  class ImageContainerBySTLMap:
     public std::map<typename TDomain::Point, TValue >
   {
 
   public:
 
-    typedef ImageContainerBySTLMap<TDomain,TValue> Self; 
-    typedef std::map<typename TDomain::Point, TValue > Parent; 
+    typedef ImageContainerBySTLMap<TDomain,TValue> Self;
+    typedef std::map<typename TDomain::Point, TValue > Parent;
 
     /// domain
     BOOST_CONCEPT_ASSERT(( concepts::CDomain<TDomain> ));
-    typedef TDomain Domain;    
+    typedef TDomain Domain;
     typedef typename Domain::Point Point;
     typedef typename Domain::Vector Vector;
     typedef typename Domain::Integer Integer;
     typedef typename Domain::Size Size;
     typedef typename Domain::Dimension Dimension;
     typedef Point Vertex;
 
-    // Pointer to the (const) Domain given at construction.
-    typedef const Domain * DomainPtr;
-
-
     /// static constants
     static const typename Domain::Dimension dimension;
 
     /// range of values
     BOOST_CONCEPT_ASSERT(( CLabel<TValue> ));
     typedef TValue Value;
-    typedef DefaultConstImageRange<Self> ConstRange; 
-    typedef DefaultImageRange<Self> Range; 
+    typedef DefaultConstImageRange<Self> ConstRange;
+    typedef DefaultImageRange<Self> Range;
 
     /// output iterator
-    typedef SetValueIterator<Self> OutputIterator; 
+    typedef SetValueIterator<Self> OutputIterator;
 
     /////////////////// Data members //////////////////
-  private: 
+  private:
 
-    /// Counted pointer on the image domain,
+    /// Shared pointer on the image domain,
     /// Since the domain is not mutable, not assignable,
     /// it is shared by all the copies of *this
-    DomainPtr myDomainPtr;
+    std::shared_ptr<const Domain> myDomainPtr;
```

Comment:
Yes it can ! However  `CountedPtr` has no conversion from `CountedPtr<T>` to `CountedPtr<const T>` that may be necessary here. I will think about adding it...